graylog2-web-interface/src/views/logic/slices/undoRedoSlice.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+import type { AppDispatch } from 'stores/useAppDispatch';
+import { updateViewNoBufferPush } from 'views/logic/slices/viewSlice';
+import type { RootState, ViewState } from 'views/types';
+import { selectRootUndoRedo, selectUndoRedoAvailability } from 'views/logic/slices/undoRedoSelectors';
+
+const BUFFER_MAX_SIZE = 10;
+
+type BufferItemType = 'view';
+
+export type BufferItem = { type: BufferItemType, state: any }
+
+export type UndoRedoState = {
+  buffer: Array<BufferItem>,
+  currentRevision: number
+}
+
+const handlers: { [name in BufferItemType]: any} = {
```

Comment:
With a single entry, we do not need this type-based dispatch. We should remove it, as we can still add it in case we really need it.